openedx/core/djangoapps/team_api/tests/test_views.py

```diff
@@ -0,0 +1,437 @@
+
+    def test_list_topics_order_by_name_by_default(self):
+        self.client.login(username=self.student_user_enrolled, password=self.test_password)
+        response = self.client.get(reverse('topics_list') + '?course_id=' + str(self.test_course_1.id))
+        self.assertEqual(200, response.status_code)
+        topics = response.data['results']
+        self.assertEqual(topics, sorted(topics, key=lambda t: t['name']))
+
+    def test_list_topics_order_by_name(self):
+        self.client.login(username=self.student_user_enrolled, password=self.test_password)
+        params = '?course_id={}&order_by=name'.format(str(self.test_course_1.id))
+        response = self.client.get(reverse('topics_list') + params)
+        self.assertEqual(200, response.status_code)
+        topics = response.data['results']
+        self.assertEqual(topics, sorted(topics, key=lambda t: t['name']))
+
+    def test_list_topics_order_by_team_count(self):
+        self.client.login(username=self.student_user_enrolled, password=self.test_password)
+        params = '?course_id={}&order_by=team_count'.format(str(self.test_course_1.id))
+        response = self.client.get(reverse('topics_list') + params)
+        self.assertEqual(200, response.status_code)
+        topics = response.data['results']
+        self.assertEqual(topics, sorted(topics, key=lambda t: t['team_count']))
+
+    def test_list_topics_invalid_ordering(self):
+        self.client.login(username=self.student_user_enrolled, password=self.test_password)
+        params = '?course_id={}&order_by=BOGUS'.format(str(self.test_course_1.id))
+        response = self.client.get(reverse('topics_list') + params)
+        self.assertEqual(400, response.status_code)
+
+    def test_list_topics_pagination(self):
+        page_size = 2
+        self.client.login(username=self.student_user_enrolled, password=self.test_password)
+        params = '?course_id={}&page_size={}'.format(str(self.test_course_1.id), str(page_size))
+        response = self.client.get(reverse('topics_list') + params)
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(page_size, len(response.data['results']))
+        self.assertIn('next', response.data)
+        self.assertIn('previous', response.data)
+        self.assertNotEqual(response.data['next'], 'null')
+
+    def test_topic_detail_anonymous(self):
+        response = self.client.get(
+            reverse('topics_detail', kwargs={'topic_id': 'topic_0', 'course_id': str(self.test_course_1.id)})
+        )
+        self.assertEqual(403, response.status_code)
+
+    def test_topic_detail_unenrolled(self):
+        self.client.login(username=self.student_user, password=self.test_password)
+        response = self.client.get(
+            reverse('topics_detail', kwargs={'topic_id': 'topic_0', 'course_id': str(self.test_course_1.id)})
+        )
+        self.assertEqual(403, response.status_code)
+
+    def test_topic_detail_invalid_course_id(self):
```

Comment:
@cahrens I'm not familiar with ddt; is there a good example of its use that I could check out?